src/app.jsx

```diff
@@ -129,34 +149,30 @@ const NavigatorCardHeader = ({ currentFilter, onFilterChange }) => {
 };
 
 const NavigatorCardBody = ({ currentFilter, files, setPath, path, pathIndex, setPathIndex }) => {
-    const onDoubleClickNavigate = (dir, path, file) => {
-        if (dir) {
-            setPath([...path.slice(0, pathIndex), file]);
-            setPathIndex(pathIndex + 1);
+    const onDoubleClickNavigate = (path, file) => {
+        if (file.type === "directory") {
+            setPath((p) => [...p, file.name]);
+            setPathIndex((p) => p + 1);
         }
     };
 
     return (
         <CardBody>
             <Flex id="folder-view">
                 {files.map((file) => {
-                    const directory = file.substring(file.length - 1) === "/";
-                    if (directory)
-                        file = file.substring(0, file.length - 1);
-
-                    if (file.toLowerCase().includes(currentFilter.toLowerCase())) {
+                    if (file.name.toLowerCase().includes(currentFilter.toLowerCase())) {
                         return (
-                            <Flex key={file} direction={{ default: "column" }} spaceItems={{ default: 'spaceItemsNone' }}>
+                            <Flex key={file.name} direction={{ default: "column" }} spaceItems={{ default: 'spaceItemsNone' }}>
                                 <FlexItem alignSelf={{ default: "alignSelfCenter" }}>
-                                    <Button variant="plain" onDoubleClick={ () => onDoubleClickNavigate(directory, path, file)}>
+                                    <Button variant="plain" onDoubleClick={ () => { onDoubleClickNavigate(path, file) }}>
```

Comment:
No need to add {} around onDoubleClickNavigate here. When you can write less it is better.